src/main/java/nl/tudelft/semgroup4/GameStateController.java

```diff
@@ -0,0 +1,93 @@
+package nl.tudelft.semgroup4;
+
+import nl.tudelft.model.Game;
+import nl.tudelft.semgroup4.logger.LogSeverity;
+import nl.tudelft.semgroup4.resources.ResourcesWrapper;
+
+import org.newdawn.slick.GameContainer;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.Input;
+import org.newdawn.slick.SlickException;
+import org.newdawn.slick.state.StateBasedGame;
+
+public class GameStateController {
+    
+    private Game currentGame;
+    
+    /**
+     * Creates a game state controller, which controls the logic for the 
+     * GameState.
+     * @param game : the current game this GameState is based on.
+     */
+    protected GameStateController(Game game) {
+        this.currentGame = game;
+    }
+    
+    /**
+     * Checks if the game is paused with the pause screen opened (which would toggle 
+     * the pausescreen to close) or if the game is running (which would toggle the 
+     * pausescreen to open.
+     * @param pauseScreenOpened : boolean that indicates if the pause screen is open.
+     * @param input : slick input. 
+     * @return : booelean that indicates if the pause screen is open.
+     */
+    protected boolean togglePauseMenu(boolean pauseScreenOpened, Input input) {
```

Comment:
Passing the <code>pauseScreenOpened</code> boolean is not necessary, you can simply ask the GameState that this controller belongs to for its state, assuming you pass that to its constructor (instead of a game).

< interface